adding x tag to nip56. (#1669)
nostr-protocol/nips: 56.md

- `impersonation` - someone pretending to be someone else
- `other` - for reports that don't fit in the above categories

Some report tags only make sense for profile reports, such as `impersonation`.

> - `x` tags SHOULD be info hash of a blob which is intended to be report. when the `x` tag is represented client MUST include an `e` tag which is the id of the event that contains the mentioned blob. also, additionally these events can contain a `server` tag to point to media servers which may contain the mentioned media.

Reviewer: ```suggestion
- `x` tag SHOULD be the hash of a blob which is being reported. Clients MAY also include an `e` or `a` tag to show where this blob is being posted.
```

Reviewer: How can you report multiple `x` tags while also using the `e` tag, `server` is useless

Reviewer: Seems unnecessary to shortcut to the url when you can lookup the url in the note via `e` tag

Author: > How can you report multiple `x` tags while also using the `e` tag, `server` is useless

for the x tag, i believe it needs to be along with bud-09. for the server part, it makes sense.

Author: any updates guys?

{
  "kind": 1984,
  "tags": [
    ["p", "<impersonator pubkey>", "impersonation"]
  ],
  "content": "Profile is impersonating nostr:<victim bech32 pubkey>",
  // other fields...
}
```

```jsonc
{
  "kind": 1984,
  "tags": [
    ["x", "<blob hash>", "malware"],
    ["e", "<event id which contains the blob on x tag>", "malware"],

>     ["server", "https://you-may-find-the-blob-here.com"]

Author: makes sense. since the full url also contains the base url. done.
